persistence/src/commonMain/db_user/migrations/26.sqm

```diff
@@ -1,31 +1,105 @@
+      text_body TEXT,
+      quoted_message_id TEXT,
+      is_quote_verified INTEGER AS Boolean,
+      is_quoting_self INTEGER AS Boolean NOT NULL,
 
       FOREIGN KEY (message_id, conversation_id) REFERENCES Message(id, conversation_id) ON DELETE CASCADE ON UPDATE CASCADE,
       PRIMARY KEY (message_id, conversation_id)
 );
 
--- Create new table for Conversation Receipt Mode Changed Content
+INSERT INTO Temp_MessageTextContent(message_id, conversation_id, text_body, quoted_message_id, is_quote_verified, is_quoting_self)
+SELECT message_id, conversation_id, text_body, quoted_message_id, is_quote_verified, (
+    CASE WHEN
+        quoted_message_id IS NULL
+            THEN 0
+            ELSE (
+                IFNULL(
+                    (SELECT (Message.sender_user_id == SelfUser.id)
+                        FROM Message
+                        LEFT JOIN SelfUser
+                     WHERE
+                        Message.id = :quoted_message_id AND
+                        conversation_id = :conversation_id),
+                0 )
+                )END
+) FROM MessageTextContent;
 
-CREATE TABLE MessageConversationReceiptModeChangedContent (
-      message_id TEXT NOT NULL,
-      conversation_id TEXT AS QualifiedIDEntity NOT NULL,
+DROP INDEX message_text_content_quoted_id_index;
+DROP TABLE MessageTextContent;
 
-      receipt_mode INTEGER AS Boolean  NOT NULL DEFAULT(0),
+ALTER TABLE Temp_MessageTextContent RENAME TO MessageTextContent;
+CREATE INDEX message_text_content_quoted_id_index ON MessageTextContent(quoted_message_id) WHERE quoted_message_id IS NOT NULL;
+CREATE INDEX message_conversation_index ON Message(conversation_id); -- this index hahe been added to the schema
 
-      FOREIGN KEY (message_id, conversation_id) REFERENCES Message(id, conversation_id) ON DELETE CASCADE ON UPDATE CASCADE,
-      PRIMARY KEY (message_id, conversation_id)
-);
+CREATE VIEW IF NOT EXISTS MessagePreview
+AS SELECT
+    Message.id AS id,
+    Message.conversation_id AS conversationId,
+    Message.content_type AS contentType,
+    Message.creation_date AS date,
+    Message.visibility AS visibility,
+    User.name AS senderName,
+    User.connection_status AS senderConnectionStatus,
+    User.deleted AS senderIsDeleted,
+    SelfUser.id AS selfUserId,
+    (Message.sender_user_id == SelfUser.id) AS isSelfMessage,
+    MemberChangeContent.member_change_list AS memberChangeList,
+    MemberChangeContent.member_change_type AS memberChangeType,
+    ConversationNameChangedContent.conversation_name AS updateConversationName,
+    Conversation.name AS conversationName,
+    (Mention.user_id IS NOT NULL) AS isMentioningSelfUser,
+    TextContent.is_quoting_self AS isQuotingSelfUser,
+    TextContent.text_body AS text,
+    AssetContent.asset_mime_type AS assetMimeType,
+    (Message.creation_date > Conversation.last_read_date) AS isUnread,
+    IFNULL((Message.creation_date > IFNULL(Conversation.last_notified_date, 0)), FALSE) AS shouldNotify,
```

Comment:
do we need this if NotificationMessage has it's own query?

Reply:
i left is just so 26.sqm matches on release and dev; otherwise we will get nasty errors when running migration